python/pyspark/pandas/frame.py

```diff
@@ -3459,6 +3458,109 @@ def mask(
         cond_inversed = cond._apply_series_op(lambda psser: ~psser)
         return self.where(cond_inversed, other)
 
+    # TODO: Support axis as 1 or 'columns'
+    def mode(self, axis: Axis, numeric_only: bool = False, dropna: bool = True) -> "DataFrame":
+        """
+        Get the mode(s) of each element along the selected axis.
+
+        The mode of a set of values is the value that appears most often.
+        It can be multiple values.
+
+        Notes
+        -----
+        The current implementation of mode requires joins multiple times
```

Comment:
If the join is expensive is there a way to just add/concatenate all of the columns to the same dataframe without joining? Since you don't care about the order of the modes anyway, the columns are more or less independent.

Reply:
The implementation of mode would still be expensive since you need to do a group by per column but at least you avoid the join when there are many modes